docs/modelserving/servingruntimes.md

```diff
@@ -0,0 +1,179 @@
+by the runtime will be used for model deployment.
+
+### Implicit: Automatic selection
+
+In each entry of the `supportedModelFormats` list, `autoSelect: true` can optionally be specified to indicate that that the given `ServingRuntime` can be
+considered for automatic selection for predictors with the corresponding model format if no runtime is explicitly specified.
+For example, the `kserve-sklearnserver` _ClusterServingRuntime_ supports SKLearn version 1 and has `autoSelect` enabled:
+
+```yaml
+apiVersion: serving.kserve.io/v1alpha1
+kind: ClusterServingRuntime
+metadata:
+  name: kserve-sklearnserver
+spec:
+  supportedModelFormats:
+    - name: sklearn
+      version: "1"
+      autoSelect: true
+...
+```
+
+When the following InferenceService is deployed with no runtime specified, the controller will look for a runtime that supports `sklearn`:
+
+```yaml
+apiVersion: serving.kserve.io/v1beta1
+kind: InferenceService
+metadata:
+  name: example-sklearn-isvc
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: sklearn
+      storageUri: s3://bucket/sklearn/mnist.joblib
+```
+
+Since `kserve-sklearnserver` has an entry in its `supportedModelFormats` list with `sklearn` and `autoSelect: true`, this _ClusterServingRuntime_
+will be used for model deployment.
+If multiple runtimes have `sklearn` as an auto-selectable supported format, then there is no guarantee _which_ runtime will be selected.
+So users and cluster-administrators should enable `autoSelect` with care.
+
+### Previous schema
+
+Currently, if a user uses the old schema for deploying predictors where you specify a framework/format as a key, then a KServe webhook will automatically map it to one of the out-of-the-box _ClusterServingRuntimes_. This is for backwards compatibility. For example:
+
+=== "Previous Schema"
+    ```yaml
+    apiVersion: serving.kserve.io/v1beta1
+    kind: InferenceService
+    metadata:
+      name: example-sklearn-isvc
+    spec:
+      predictor:
+        sklearn:
+          storageUri: s3://bucket/sklearn/mnist.joblib
+    ```
+=== "Equivalent New Schema"
+    ```yaml
+    apiVersion: serving.kserve.io/v1beta1
+    kind: InferenceService
+    metadata:
+      name: example-sklearn-isvc
+    spec:
+      predictor:
+        model:
+          modelFormat:
+            name: sklearn
+          storageUri: s3://bucket/sklearn/mnist.joblib
+          runtime: kserve-sklearnserver
+    ```
+
+The previous schema would mutate into the new schema where the `kserve-sklearnserver` _ClusterServingRuntime_ is explicitly specified.
+
+> **Note**: The old schema will eventually be removed in favor of the new Model spec, where a user can specify a model format and optionally a corresponding version.
+
+In previous versions of KServe, supported predictor formats and container images were defined in a
+[config map](https://github.com/kserve/kserve/blob/release-0.7/config/configmap/inferenceservice.yaml#L7) in the control plane namespace.
```

Comment:
We found that we still need to keep the configmap as for existing inference service upgraded from 0.7 to 0.8, during reconciliations they do not go through the mutating webhook, so it executes the old code path(see the code [here](https://github.com/kserve/kserve/blob/master/pkg/controller/v1beta1/inferenceservice/components/predictor.go#L151)).

Reply:
Interesting. I adjusted the wording here to note its continued use for upgraded InferenceServices.